.github/workflows/docker.yml

```diff
@@ -0,0 +1,122 @@
+name: Docker
+
+on:
+  push:
+    branches:
+      - 'master'
+  release:
+    types: [published]
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: checkout repo
+        uses: actions/checkout@v2
+
+      - name: build and push docker-image
+        id: date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+
+      - name: build and push docker-image
+        id: gen_tags
+        run: |
+          if [[ $GITHUB_REF == 'refs/tags/'* ]]; then
+            TAGS='["latest","'${GITHUB_REF/refs\/tags\//}'"]'
+          else
+            TAGS='["'${GITHUB_SHA}'_'${{ steps.date.outputs.date }}'"]'
+          fi
+          echo '::set-output name=tags::'$TAGS
+
+      - name: build and push docker-image
+        if: github.event.push
+        run: make docker-build-push
+        env:
+          PROCESS: persistencecore
+          DOCKER_IMAGE_NAME: persistenceone/persistencecore
+          DOCKER_TAG_NAME: ${{ steps.gen_tags.outputs.tags }}
+
+      - name: build and push docker-image for release
+        if: github.event.release
+        run: make docker-build-push
+        env:
+          PROCESS: persistencecore
+          DOCKER_IMAGE_NAME: persistenceone/persistencecore
+          DOCKER_TAG_NAME: ${{ steps.gen_tags.outputs.tags }}
```

Comment:
does this even work?

Reply:
Generating the flags like this? Yes it works

Reply:
This is how the tag looks like `persistenceone/persistencecore:[20220919_1c287]`